kubernetes/deployments_kubeclient.go

```diff
@@ -894,13 +924,136 @@ func (oc *openShiftAPIClient) GetDeploymentConfig(namespace string, name string)
 	return oc.getResource(dcURL, true)
 }
 
-func (kc *kubeClient) deleteDeploymentConfig(spaceName string, appName string, namespace string) error {
+const buildConfigLabelName = "openshift.io/build-config.name"
+const envServicesAnnotationPrefix = "environment.services.fabric8.io"
+const envServicesDeploymentVersions = "deploymentVersions"
+
+func (kc *kubeClient) getDeploymentConfigNameForApp(namespace string, appName string, spaceName string) (string, error) {
+	// Look up builds with config name matching appName, this will be the one created by the launcher
+	labelSelector := fmt.Sprintf("%s=%s,%s=%s", buildConfigLabelName, appName, spaceLabelName, spaceName)
+	escapedSelector := url.QueryEscape(labelSelector)
+
+	// Builds are located in the user's namespace of type "user"
+	resp, err := kc.GetBuilds(kc.config.UserNamespace, escapedSelector)
```

Comment:
Question: if there are no builds, or there's some other failure, what would be the consequence of falling back to returning appName?

Reply:
It would be no different than what we have now. In the worst case, the user (for some reason) has a Deployment Config and Build Config with the same name, but they are not actually related. If we default to using the appName, we'll show them as related.

I think it may still be a good idea to use appName as a fallback. It would mean that at the very minimum, this PR will maintain the existing level of functionality.

Reply:
I'll leave it up to your judgement.  I'm good with this either way (I slightly favour falling back), but I want to ensure the code doesn't introduce new failure modes.